backends/candle/src/models/debertav2.rs

```diff
@@ -0,0 +1,1490 @@
+        mask: Option<&Tensor>,
+        inputs_embeds: Option<&Tensor>,
+    ) -> Result<Tensor> {
+        let (input_shape, input_embeds) = match (input_ids, inputs_embeds) {
+            (Some(ids), None) => {
+                let embs = self.word_embeddings.forward(ids)?;
+                (ids.dims(), embs)
+            }
+            (None, Some(e)) => (e.dims(), e.clone()),
+            (None, None) => {
+                bail!("Must specify either input_ids or inputs_embeds")
+            }
+            (Some(_), Some(_)) => {
+                bail!("Can't specify both input_ids and inputs_embeds")
+            }
+        };
+
+        let seq_length = match input_shape.last() {
+            Some(v) => *v,
+            None => bail!("DebertaV2Embeddings invalid input shape"),
+        };
+
+        let position_ids = match position_ids {
+            Some(v) => v.clone(),
+            None => self.position_ids.narrow(1, 0, seq_length)?,
+        };
+
+        let token_type_ids = match token_type_ids {
+            Some(ids) => ids.clone(),
+            None => Tensor::zeros(input_shape, DType::U32, &self.device)?,
+        };
+
+        let position_embeddings = match &self.position_embeddings {
+            Some(emb) => emb.forward(&position_ids)?,
+            None => Tensor::zeros_like(&input_embeds)?,
+        };
+
+        let mut embeddings = input_embeds;
+
+        if self.config.position_biased_input {
+            embeddings = embeddings.add(&position_embeddings)?;
+        }
+
+        if self.config.type_vocab_size > 0 {
+            embeddings = self.token_type_embeddings.as_ref().map_or_else(
+                || bail!("token_type_embeddings must be set when type_vocab_size > 0"),
+                |token_type_embeddings| {
+                    embeddings.add(&token_type_embeddings.forward(&token_type_ids)?)
+                },
+            )?;
+        }
+
+        if self.embedding_size != self.config.hidden_size {
+            embeddings = if let Some(embed_proj) = &self.embed_proj {
+                embed_proj.forward(&embeddings)?
+            } else {
+                bail!("embed_proj must exist if embedding_size != config.hidden_size");
+            }
+        }
+
+        embeddings = self.layer_norm.forward(&embeddings)?;
+
+        if let Some(mask) = mask {
+            let mut mask = mask.clone();
+            if mask.dims() != embeddings.dims() {
+                if mask.dims().len() == 4 {
+                    mask = mask.squeeze(1)?.squeeze(1)?;
+                }
+                mask = mask.unsqueeze(2)?;
+            }
+
+            mask = mask.to_dtype(embeddings.dtype())?;
+            embeddings = embeddings.broadcast_mul(&mask)?;
+        }
+
+        Ok(embeddings)
+    }
+}
+
+// https://github.com/huggingface/transformers/blob/78b2929c0554b79e0489b451ce4ece14d265ead2/src/transformers/models/deberta_v2/modeling_deberta_v2.py#L72
+struct XSoftmax {}
+
+impl XSoftmax {
```

Comment:
I guess, we don't need this layer! Just add the attention mask to the input like [this](https://github.com/huggingface/text-embeddings-inference/blob/6e900afba71821fdf250e380d7da1f5a6e5e7e27/backends/candle/src/models/modernbert.rs#L261)